Data split into training (915 samples), validation (114 samples), and testing (114 samples) sets.
Number of classes: 6
Feature data normalized using z-score normalization.
Training:   0%|                                                                                                                                                                 | 0/1000 [00:00<?, ?epoch/s]/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py:246: RuntimeWarning: divide by zero encountered in log
  return -np.mean(self.y*np.log(self.y_pred) + (1-y)*np.log(1-self.y_pred))
/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py:248: RuntimeWarning: divide by zero encountered in divide
  return -((self.y/(self.y_pred)) + ((1-self.y)/(1-self.y_pred)))
/home/autrio/.local/lib/python3.10/site-packages/numpy/core/_methods.py:118: RuntimeWarning: invalid value encountered in reduce
  ret = umr_sum(arr, axis, dtype, out, keepdims, where=where)
/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py:393: RuntimeWarning: invalid value encountered in multiply
  dLy = dLy * daz
/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py:392: RuntimeWarning: invalid value encountered in matmul
  dLy = dLy @ self.layers[i + 1].W
Training:   9%|██████████▊                                                                                                               | 89/1000 [00:22<03:53,  3.90epoch/s, Train Acc=0.00781, Val Acc=0]
Epoch: 0 	Train:[Loss: nan, Acc: 0.0742] 	Val:[Loss: nan, Acc: 0.0000]
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/assignments/3/2/2/2.py", line 51, in <module>
    run(setup())
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/assignments/3/2/2/2.py", line 45, in run
    model.train(epochs=params["epoch"])
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 437, in train
    train_loss, train_acc = self.train_step(train_loader)
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 415, in train_step
    self.backward()
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 390, in backward
    dzw, dzx, daz = self.layers[i].get_grads()
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 288, in get_grads
    grads = [np.stack(self.dzw, axis=1),
  File "/home/autrio/.local/lib/python3.10/site-packages/numpy/core/shape_base.py", line 456, in stack
    return _nx.concatenate(expanded_arrays, axis=axis, out=out,
KeyboardInterrupt
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/assignments/3/2/2/2.py", line 51, in <module>
    run(setup())
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/assignments/3/2/2/2.py", line 45, in run
    model.train(epochs=params["epoch"])
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 437, in train
    train_loss, train_acc = self.train_step(train_loader)
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 415, in train_step
    self.backward()
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 390, in backward
    dzw, dzx, daz = self.layers[i].get_grads()
  File "/home/autrio/college-linx/SMAI/smai-m24-assignments-Autrio/models/MLP/MLP2.py", line 288, in get_grads
    grads = [np.stack(self.dzw, axis=1),
  File "/home/autrio/.local/lib/python3.10/site-packages/numpy/core/shape_base.py", line 456, in stack
    return _nx.concatenate(expanded_arrays, axis=axis, out=out,
KeyboardInterrupt
